src/guiguts/footnotes.py

```diff
@@ -122,28 +143,831 @@ def join_to_previous(self) -> None:
             f"{fn_cur_start} -1l linestart", f"{fn_cur_end} +1l linestart"
         )
         maintext().delete(f"{fn_prev_end} -1c", f"{fn_prev_end} lineend")
-        maintext().insert(fn_prev_end, "\n" + continuation_text + "\n")
+        maintext().insert(fn_prev_end, "\n" + continuation_text)
         self.checker_dialog.remove_entry_current()
-        maintext().see(fn_prev_end)
+        # Note index of the record to which we joined the continuation text.
+        saved_prev_rec_index = fn_index - 1
+        self.run_check()
+        display_footnote_entries(auto_select_line=False)
+        fn_records = self.get_fn_records()
+        prev_record = fn_records[saved_prev_rec_index]
+        # Get index into dialog entries for this record.
+        dialog_entry_index = self.map_fn_record_to_dialog_index(prev_record)
+        if dialog_entry_index < 0:
+            logger.error("Unexpected return from 'join_to_previous()'")
+            return
+        # Make it the selected entry in the dialog.
+        self.checker_dialog.select_entry_by_index(dialog_entry_index)
+
+    def autoset_chapter_lz(self) -> None:
+        """Insert a 'FOOTNOTES:' LZ header line before each chapter break.
```

Comment:
Discrepancy between GG2 and GG1 (moving footnotes to chapters).

In the first screenshot. Notice the position of fn 1 in the text file (split screen is irrelevant in this screenshot, ignore fn 2).

![fn-lz1](https://github.com/user-attachments/assets/8f55ab0f-977b-4ca7-8661-f3265ff8fb8e)

GG1 does not place an LZ following the chapter title, but above the following chapter header.


I'm not sure what the difference is (I may need to look at the file more closely later), but notice that in this second screenshot, the footnote anchor is in the same place, but in this case, the footnote is more correctly moved to the end of the chapter:

<img width="600" alt="image" src="https://github.com/user-attachments/assets/59ff4f35-6c59-48df-91da-2c00cb8c650b">

---

A couple of comments/questions.

1. I understand that GG1 places the landing zones roughly the same way, but to me, looking at where the landing zones are placed, they're not where I would expect them to be. That is, yes, they're placed before the chapter header by four blank lines, but, if the chapters all start on a new page, that means that the footnotes for one chapter are actually on the same page as the beginning of the following chapter. Which seems to me to be problematical if a file has page numbers associated with it. Am I missing something that's a part of PP'ing?
2. I expected "Tidy Footnotes" to remove the unwanted "FOOTNOTES:" tags. It didn't for me in GG2. I checked GG1, and it did. Is this a Mac issue, or is it again something I should be doing that I'm not aware of?

Reply:
> A couple of comments/questions.
> 
> 1. I understand that GG1 places the landing zones roughly the same way, but to me, looking at where the landing zones are placed, they're not where I would expect them to be. That is, yes, they're placed before the chapter header by four blank lines, but, if the chapters all start on a new page, that means that the footnotes for one chapter are actually on the same page as the beginning of the following chapter. Which seems to me to be problematical if a file has page numbers associated with it. Am I missing something that's a part of PP'ing?
> 
> 2. I expected "Tidy Footnotes" to remove the unwanted "FOOTNOTES:" tags. It didn't for me in GG2. I checked GG1, and it did. Is this a Mac issue, or is it again something I should be doing that I'm not aware of?

After touching base with @windymilla, I learned a couple of things:

* Regarding the unused LZ removal, that should happen when the footnotes are moved to LZs.

* Tidy Footnotes is only supposed to be used on the text file, after Text and HTML are split.

So, I was indeed misunderstanding. However, the unused LZs should be removed.

Reply:
Re. removing unused LZ headers, the current (and new) Guiguts Manual agrees with you and reminds PPers that:

"If you used **Autoset Chap. LZ**, there may be unused Landing Zones at the ends of chapters with no footnotes. (Recall that Landing Zones are the word FOOTNOTES:). You probably will want to delete the unused Landing Zones. Similarly, if you moved all of the footnotes to the end of the document and later moved them to follow the paragraphs that referenced them, remember to delete FOOTNOTES: at the end of the document."

Reply:
@srjfoo and you are right about this. I'll add that fix along with the other issue that @srjfoo identified.

Reply:
@G4OEU -- many thanks. I just want to let you know that your effort is very much appreciated! Filling out Footnote Fixup is a major undertaking!

It's probably both an advantage and a disadvantage as a tester, that my perspective is that of someone who has done some post-processing, long ago, but not enough to remember the whole procedure and what did and didn't work in GG1 in the days of 0.2.4. I'll let @windymilla pass judgment on that (😁). It does encourage me to ask questions when something doesn't make sense to me, though!